ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java

```diff
@@ -243,8 +243,8 @@ public void stop() {
     stop(false);
```

Comment:
One thing though -- can we make sure that we're still resetting the offsets if the query is actually explicitly terminated? I'm worried that we aren't doing that anymore.

 I think we should keep the default behavior of `stop` to mean "clean up the offsets" ie change this to true, but just make sure to pass in false when we're actually just stopping/pausing them in `QueryRegistryImpl` (ie [here](https://github.com/confluentinc/ksql/pull/9407/files#diff-3d6026517bf65fbc5c4619defec666b7a5261137ba02a69a66604137f84f47f3R461), as you're already doing)

Reply:
Another thing that's a bit too confusing as it is now but hopefully we can clean up in a later pass. Too many different meanings of `stop` and `close` floating around

Reply:
Yeah we have a unit test for that, and yeah I can keep the default to clean it up